Bot Scraping Functions & Asking for Title

ItsNotJiaching opened the pull request:
all of the bot functions we've been creating up until now. the bot asks for the title of the documentation and scrapes the message, but a bug is that it'll scrape all reply messages, not just the one message intended to be the title of the wiki documentation.

ItsNotJiaching committed:
code as of 8/8/22

bot reacts on a certain emoji only and will only write the message being reacted to in the files.

ItsNotJiaching committed:
let bot ask for title & scrapes replies

Added messages.txt, requirements.txt, and onboarding_tutorial.py.

Slackbot will now ask user for a title for the documentation message they've reacted to. They will also scrape the user's title message if it was sent within 10 minutes of the message being sent. This satisfies DS-16.

The retrieval of the timestamp of a message that received a reaction has been fixed so that it longer outputs None. This fixes DS-10.

Closed Issues: DS-10 and DS-16 on Jira

Current issues:
1. Slackbot will scrape all replies to a message.
2. The 10-min timer is from when the message was SENT, not when the message was replied to.
These issues will be resolved in the next commit, and its corresponding issue ticket will be DS-17.

iengel35 commented on onboarding_tutorial.py:
```diff
@@ -0,0 +1,77 @@
+class OnboardingTutorial:
```
Please add a comment at the top of this file linking to the tutorial you got it from

ItsNotJiaching replied:
The comment has been added.

#slackbot tutorial original code from https://github.com/slackapi/python-slack-sdk/tree/main/tutorial, modified.

iengel35 commented on messages.txt:
```diff
@@ -0,0 +1 @@
+tseting again!
```
You generally don't want to commit files generated at runtime to a repository- you can add `*.txt` (and `*.md`) to your `.gitignore` to avoid accidentally committing this type of file. (https://git-scm.com/docs/gitignore)

ItsNotJiaching replied:
As of right now, if the messages.txt file does not exist, when the code runs it will cause an error when it eventually attempts to read a file. Should I remove it for now, and then in the next commit, I add in supplemental code that'll create a file if there isn't one already?

iengel35 replied:
If you use `+a` instead of `+r` when opening, the file will be created if it does not already exist: https://stackoverflow.com/questions/2967194/open-in-python-does-not-create-a-file-if-it-doesnt-exist

Going forward, you might want to write a function which does your file opening/writing and you can just pass text into it. That way, your code will be simpler to read, and if you change the way you do file writing, you will only need to change it in one place!

ItsNotJiaching replied:
Alright, I will put the function into consideration. For now I've updated the code to use a+.

iengel35 commented on app.py:
```diff
@@ -0,0 +1,172 @@
+def message(event, client):
+    # initializes the file for the messages and puts cursor at the last line
+    file = open("PythOnBoardingBot/messages.txt", "r+")
+    file.seek(len(file.read(-1)))
```
You open the file later in this function, looks like you only need to keep one of them

ItsNotJiaching replied:
Oops- small mistake while copying & pasting code. Lines 137-139 have been deleted.

iengel35 commented on app.py:
```diff
@@ -0,0 +1,172 @@
+import logging
```
Please also add a link to this file with the tutorial code you used

ItsNotJiaching replied:
added! 

#starting code from https://github.com/slackapi/python-slack-sdk/tree/main/tutorial

iengel35 commented on app.py:
```diff
@@ -0,0 +1,172 @@
+    # it'll be written into messages.txt as the title of documentation. Will fix later.
+    try:    
+        print(float(message_ts) - float(message_thread_ts))
+        if float(message_ts) - float(message_thread_ts) <= 600:
```
Since each page will only have one title, and each section only one heading, have you considered storing whether you have already scraped a title from that thread instead of using time?

ItsNotJiaching replied:
Sorry for not having responded to this before I started working on changing how the bot scrapes messages. I did take the idea and altered it slightly to have the code use the timestamp of a message to check whether the message has been scraped already. This was a much bigger change than just a few lines. Do I need to update the code in this PR specifically or can I put the updated code in the next PR?

iengel35 replied:
Feel free to stick this in another PR! Just was throwing out an idea- will mark this as resolved.

iengel35 commented:
Thanks for opening a PR, overall nice job writing clear code and commenting it effectively- made a couple minor comments.

iengel35 reviewed (commented):
Thanks for opening a PR, overall nice job writing clear code and commenting it effectively- made a couple minor comments.

ItsNotJiaching committed:
small updates to "let bot ask for title & scrapes replies"

Below is a list of small changes to the code:

1. added references to the tutorial code used in onboarding_tutorial.py and app.py
2. removed redundant file-opening code in app.py. This does not change any functionality.
3. changed the file opening mode from r+ to a+ and removed the file.seek() line right after. a+ appends to the file if it exists, and if it doesn't, will make the file.
4. added .gitignore such that it will ignore the .txt and .md files when committing.
5. removed messages.txt

ItsNotJiaching committed:
small updates to "let bot ask for title & scrapes replies" (addendum)

change 3 is now applied in line 80 as well.

iengel35 reviewed (approved):
LGTM now!
